concert/base.py

```diff
@@ -860,3 +861,61 @@ def unlock(self):
         """Unlock all the parameters for writing."""
         for param in self:
             param.unlock()
+
+
+_INSTANCE_CACHE = False
+
+
+def _memoize_instance_id(func):
```

Comment:
Just wondering, but is there a very urgent reason to implement that behaviour as a decorator?